### PR TITLE
Add preserve option

### LINKED ISSUE #32: Support the --preserve option
Hi
First of all, thank you for this project, I feel it could help me re-building more efficient bundles, if I managed to have it work completely :)

### Expected behaviour
Trying to have the files with the correct directory structure. The `--preserve` option is documented but fails

```
> npx shuji -v  --output-dir OUT_SHUJI $npm_package_config_map --preserve
Invalid option '--preserve' - perhaps you meant '--verbose'?
```

I checked the code, and indeed this option does not exist :)

Is there another way to have the original files in the original directory structure ?

thank you for your support

### PR DESCRIPTION
**Pull request by mazamachi**
Add --preserve option which enables to preserve original folder structure.
Fix #32 .

I have a question.
This worked well in my case, but I don't know if this works well for general cases 🤔 

Should I add any test cases?
Or, can I merge this PR without tests and fix this after other people report non-working cases?

Anyway, thank you for this project. This saved my day very much!

### CODE REVIEW

File: lib/write-sources.js [hunk 1]
```diff
@@ -22,6 +23,12 @@ const writeSources = (filename, content, outdir, options) => {
 
   const outputFilepath = path.join(outdir, filename);
 
+  if (options.preserve) {
+    mkdirp.sync(
```
Comment (paazmaya) [inline lib/write-sources.js]: Use https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md, so that you don't need to add new dependency

**Comment by paazmaya** [comment]
Please add test cases

**Comment by Sapphire2k** [comment]
it works!

**Comment by rodrigograca31** [comment]
@mazamachi can you please make the requested change?

Let me know if you can't or don't have the time and I will do a new PR with it fixed.

**Comment by paazmaya** [comment]
Now released in `v0.8.0` via #39

**Pull request closed without merging**